### DISCUSSION TITLE
Tree node search function

### DISCUSSION THREAD

**Discussion by Vitor-Chiara** [Textualize/textual#5377] [Q&A]
How would i go about making a search function for tree nodes, i've got a massive tree that involves a bunch of content from api requests and i'd like to know how could i make a search bar for it, the tree code itself is simple with several roots with 4 leaves underneath it, i wanted to know how could i make a search bar for searching the roots themselves

**Comment by BurnoutDV** [comment]
You can actually iterate through all leaves/nodes by line number or ID. Its not exactly what you are asking but I was thinking aloud into the void in this thread: https://github.com/Textualize/textual/discussions/5315
